source/core/src/main/com/csse3200/game/components/npc/ShipEaterAnimationController.java

```diff
@@ -0,0 +1,49 @@
+package com.csse3200.game.components.npc;
+
+import com.csse3200.game.components.Component;
+import com.csse3200.game.rendering.AnimationRenderComponent;
+
+import java.util.Objects;
+
+public class ShipEaterAnimationController extends Component {
+	private boolean isDigging;
+	private boolean isHiding;
+	private boolean isEating;
+	private AnimationRenderComponent animator;
+	@Override
+	public void create() {
+		animator = this.entity.getComponent(AnimationRenderComponent.class);
+
+		entity.getEvents().addListener("eatingUpdated", this::setEating);
+		entity.getEvents().addListener("diggingUpdated", this::setDigging);
+		entity.getEvents().addListener("hidingUpdated", this::setHiding);
+	}
+	void setDigging(boolean isDigging) {
+		this.isDigging = isDigging;
+	}
+
+	void setEating(boolean isEating) {
+		this.isEating = isEating;
+	}
+
+	void setHiding(boolean isHiding) {
+		this.isHiding = isHiding;
+	}
```

Comment:
If only one can happen you could probs squash to a String or enum, idk if it is more efficient but decreases lines / complexity

Reply:
no way am I decreasing lines now